src/soter/boringssl/soter_ec_key.c

```diff
@@ -288,7 +292,8 @@ soter_status_t soter_ec_pub_key_to_engine_specific(const soter_container_hdr_t*
         return SOTER_INVALID_PARAMETER;
     }
 
-    if (key_length < ec_pub_key_size(curve)) {
+    /* Encoded public key cannot be smaller than this */
+    if (key_length < ec_pub_key_size(curve, true)) {
```

Comment:
Being spoiled with other languages, I would really like to see here and in other places something like `ec_pub_key_size(curve, COMPRESSED))` instead of `ec_pub_key_size(curve, true))`. But it's C, I understand :)

Reply:
in C we can use MACROS like `#define COMPRESSED true`

Reply:
I looked into changing this for some

```c
enum soter_public_key_format {
    SOTER_PUBLIC_KEY_UNCOMPRESSED,
    SOTER_PUBLIC_KEY_COMPRESSED,
}
```

and it turns out to be quite bothersome to thread this enum into all places that need it.

So... how about ~~never~~ later? 🙏

Soter is not _really_ a public API, so I'm way more open to the idea of breaking API compatibility there if necessary. ABI compatibility can be retained as well provided the enum is defined like above (0 = uncompressed, 1 = compressed).

Reply:
but why do not use simple macros just to increase the readability of these function calls? without complicating with enums

Reply:
```
bool compressed = true;

if (key_length < ec_pub_key_size(curve, compressed)) {
    // ...
}
```
I like this one